ReactAndroid/src/main/java/com/facebook/react/animated/InterpolationAnimatedNode.java

```diff
@@ -12,6 +12,32 @@
  */
 /*package*/ class InterpolationAnimatedNode extends ValueAnimatedNode {
 
+  /* package */ enum ExtrapolateType {
+    IDENTITY("identity"),
+    CLAMP("clamp"),
+    EXTEND("extend");
+
+    private final String mName;
+
+    ExtrapolateType(String name) {
+      mName = name;
+    }
+
+    public static ExtrapolateType fromString(String name) {
+      for (ExtrapolateType type : ExtrapolateType.values()) {
+        if (type.toString().equalsIgnoreCase(name)) {
+          return type;
+        }
+      }
+      throw new IllegalArgumentException("Unsupported extrapolate type : " + name);
```

Comment:
You may want to throw `JSApplicationIllegalArgumentException` here instead as this may only happen when unexpected value from the JS will get sent